packages/expo-sqlite/android/src/main/java/expo/modules/sqlite/SQLiteModule.kt

```diff
@@ -0,0 +1,326 @@
+        val rowsAffected = if (insertId >= 0) 1 else 0
+        SQLitePluginResult(EMPTY_ROWS, EMPTY_COLUMNS, rowsAffected, insertId, null)
+      } else if (isDelete(sql) || isUpdate(sql)) {
+        val rowsAffected = statement.executeUpdateDelete()
+        SQLitePluginResult(EMPTY_ROWS, EMPTY_COLUMNS, rowsAffected, 0, null)
+      } else {
+        // in this case, we don't need rowsAffected or insertId, so we can have a slight
+        // perf boost by just executing the query
+        statement.execute()
+        EMPTY_RESULT
+      }
+    } finally {
+      statement?.close()
+    }
+  }
+
+  // do a select operation
+  private fun doSelectInBackgroundAndPossiblyThrow(
+    sql: String,
+    bindArgs: Array<String?>,
+    db: SQLiteDatabase?
+  ): SQLitePluginResult {
+    var cursor: Cursor? = null
+    return try {
+      cursor = db!!.rawQuery(sql, bindArgs)
+      val numRows = cursor.count
+      if (numRows == 0) {
+        return EMPTY_RESULT
+      }
+      val numColumns = cursor.columnCount
+      val rows: Array<Array<Any?>?> = arrayOfNulls(numRows)
+      val columnNames = cursor.columnNames
+      var i = 0
+      while (cursor.moveToNext()) {
+        val row = arrayOfNulls<Any>(numColumns)
+        for (j in 0 until numColumns) {
+          row[j] = getValueFromCursor(cursor, j, cursor.getType(j))
+        }
+        rows[i] = row
+        i++
+      }
+      SQLitePluginResult(rows, columnNames, 0, 0, null)
+    } finally {
+      cursor?.close()
+    }
+  }
+
+  private fun getValueFromCursor(cursor: Cursor?, index: Int, columnType: Int): Any? {
+    when (columnType) {
+      Cursor.FIELD_TYPE_FLOAT -> return cursor!!.getDouble(index)
+      Cursor.FIELD_TYPE_INTEGER -> return cursor!!.getLong(index)
+      Cursor.FIELD_TYPE_BLOB ->
+        // convert byte[] to binary string; it's good enough, because
+        // WebSQL doesn't support blobs anyway
+        return String(cursor!!.getBlob(index))
+      Cursor.FIELD_TYPE_STRING -> return cursor!!.getString(index)
+    }
+    return null
+  }
+
+  @Throws(IOException::class)
+  private fun pathForDatabaseName(name: String): String {
+    val directory = File(mContext.filesDir.toString() + File.separator + "SQLite")
+    ensureDirExists(directory)
+    return directory.toString() + File.separator + name
+  }
+
+  @Throws(IOException::class)
+  private fun getDatabase(name: String): SQLiteDatabase? {
+    var database: SQLiteDatabase? = null
+    val path = pathForDatabaseName(name)
+    if (File(path).exists()) {
+      database = DATABASES[name]
+    }
+    if (database == null) {
+      DATABASES.remove(name)
+      database = SQLiteDatabase.openOrCreateDatabase(path, null)
+      DATABASES[name] = database
+    }
+    return database
+  }
+
+  private class SQLitePluginResult(
+    val rows: Array<Array<Any?>?>,
+    val columns: Array<String?>,
+    val rowsAffected: Int,
+    val insertId: Long,
+    val error: Throwable?
+  )
```

Comment:
It can be converted into a data class: https://kotlinlang.org/docs/data-classes.html

Reply:
What is the benefit of it in this case?